pyccel/codegen/printing/ccode.py

```diff
@@ -1325,9 +1325,22 @@ def _print_For(self, expr):
             step  = self._print(expr.iterable.step )
         else:
             raise NotImplementedError("Only iterable currently supported is Range")
-        return ('for ({target} = {start}; {target} < {stop}; {target} += '
+
+        test_step = expr.iterable.step
+        if isinstance(test_step, PyccelUnarySub):
+            test_step = expr.iterable.step.args[0]
+
+        # testing if the step is a value or an expression
+        if isinstance(test_step, Literal):
+            op = '>' if isinstance(expr.iterable.step, PyccelUnarySub) else '<'
+            return ('for ({target} = {start}; {target} {op} {stop}; {target} += '
+                        '{step})\n{{\n{body}\n}}').format(target=target, start=start, op=op,
+                                                          stop=stop, step=step, body=body)
+        else:
+            return (
+                'for ({target} = {start}; (({step} > 0 ? 1 : -1)*({target})) < (({step} > 0 ? 1 : -1)*({stop})); {target} += '
                 '{step})\n{{\n{body}\n}}').format(target=target, start=start,
-                stop=stop, step=step, body=body)
+                                                  stop=stop, step=step, body=body)
```

Comment:
Is it really needed to print this complicated exit condition? I think that this will work in C:
```c
for (i = start; step > 0 ? (i < stop) : (i > stop); i += step)
```

Reply:
By the way, can we use the name `index` instead of `target`?

Reply:
yes, I ll do that and I ll edit the for loop for C

Reply:
About the target variable I saw that [here](https://github.com/pyccel/pyccel/blob/2db510aef750bcfd59c61e62f400edbdd1e616bf/pyccel/ast/core.py#L1297-L1308) we defined the index iterator as `target` that is why I saw it was better to stay aligned with the same vocabulary. Do I still change the name from `target`  to `index`

Reply:
Actually I keep asking for using `index` instead of `target` every time I can.

I am not 100% sure about the `For` class is `ast.core`, but here `target` is an index and so the name choice is obvious to me.

Reply:
To be totally rigorous, we may want to call this "loop counter". But definitely not "target".

Reply:
In general `target` is not necessarily an `index`. E.g:
```
for v in arr:
```
The target is `v`. I'm not sure `target` is the clearest name, but I'm not sure `index` is more accurate generally. Here it is accurate as we are handling a `PythonRange`

Reply:
To summarize, I think the best name here is `counter`. That should make everybody happy! :wink:

Reply:
Understood

Reply:
done in [251cfc2](https://github.com/pyccel/pyccel/pull/780/commits/251cfc2b9b43bb5df1b23f1d6b02d6f007a6f26f)

Reply:
Thanks @abkhaoula. But have you changed the loop condition as I suggested at the beginning of this discussion? It does not appear so. As a reminder:
```python
for (counter = start; step > 0 ? (counter < stop) : (counter > stop); counter += step)
```